### PR TITLE
feat(scripts): add default config generator

### PR DESCRIPTION
**Pull request by becseya**
This PR adds a python script that can be used to update the lv_conf.h file of an existing project based on "lv_conf.default" parameters.

Do not squash please.

### CODE REVIEW

File: scripts/generate_lv_conf.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    return args
+
+
+def parse_defaults(path: str):
+    defaults = {}
+
+    with open(path, 'r', encoding='utf-8') as file:
+        for line in file.readlines():
+            groups = re.search(r'([A-Z0-9_]+)\s+(.+)', line).groups()
```
Comment (XuNeo) [inline scripts/generate_lv_conf.py:65]: May need to check basic error like empty lines.

File: scripts/generate_lv_conf.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        elif 'Set it to "1" to enable content' in src_line:
+            dst_line = '#if 1 /*Content enable*/'
+        else:
+            dst_line = src_line
+
+        dst_lines.append(dst_line)
+
+    if len(keys_used) != len(defaults):
+        fatal('The following keys are deprecated:\n  ' +
+              '\n  '.join(get_missing_keys(keys_used, defaults.keys())))
```
Comment (XuNeo) [inline scripts/generate_lv_conf.py]: Does it make sense to simply use
```suggestion
              '\n  '.join([k for k in defaults.keys() if k not in keys_used]))
```

**Review by XuNeo — DISMISSED** [review]

File: scripts/generate_lv_conf.py (hunk 1 above)
Comment (becseya) [reply scripts/generate_lv_conf.py:65]: Good catch

**Comment by kisvegabor** [comment]
Please rebase on master to fix the CI issue.

**Review by kisvegabor — DISMISSED** [review]

**Comment by kisvegabor** [comment]
@XuNeo could you take a look at this PR?

**Comment by XuNeo** [comment]
Sorry for the delay. Last time I tested, it has some problem, I can try to reproduce it.

In the meanwhile, I discussion with some fellows, and question why not use Kconfig to generate the config files?
It has standard tool and we have full support with Kconfig file.

**Comment by kisvegabor** [comment]
Kconfig has some important limitations. E.g. we can't handle [these](https://github.com/lvgl/lvgl/blob/master/lv_conf_template.h#L47-L52).

**Comment by becseya** [comment]
@XuNeo please try to reproduce.

Is Kconfig able to cover our use case? I'm not aware of a mode where we are able to generate a "full list" of the macro definitions with some default values being overwritten.

**Comment by vwheeler63** [comment]
@kisvegabor @XuNeo @becseya 

Hi, Everyone!

We are going to need to coordinate.  @becseya , I see your new indents in `lv_conf_template.h`.  In PR #6480, I was going through `lv_conf_template.h` to re-work the comments and I found the missing indent in the nested `#define`s inside `#if LV_USE_DRAW_VG_LITE`, but I missed the correct indents you made inside `#if LV_USE_GIF`.  So I added them to the one I am submitting with extensive changes.  Thank you for that.  And since I also did some minor modifications to `lv_conf_internal_gen.py` that generates `lv_conf_internal.h` (which carries the new indents forward), please allow me to submit these first, as I think you will find the changes you made to `lv_conf_template.h` and `lv_conf_internal.h` will be taken care with PR #6673 (the next PR for issue #6480).  If you are comfortable doing a rebase based on `master` after that PR is merged, I think you will find only your Python script will be needed, as the other 2 files new indents will be included therein.

Kind regards,
Vic

File: scripts/generate_lv_conf.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    if not args.defaults:
+        args.defaults = os.path.join(args.target, "lv_conf.defaults")
+
+    if not os.path.exists(args.template):
+        fatal(f"Template file not found at {args.template}")
+    if not os.path.exists(args.config):
+        fatal(f"User config file not found at {args.config}")
+    if not os.path.exists(args.defaults):
+        fatal(f"User defaults not found at {args.default}")
```
Comment (liamHowatt) [inline scripts/generate_lv_conf.py]:
```suggestion
        fatal(f"User defaults not found at {args.defaults}")
```

**Review by liamHowatt — COMMENTED** [review]
It's working for me

**Comment by XuNeo** [comment]
> @XuNeo please try to reproduce.


Sorry for the late response. I just tried the tools now and it works for me too.

**Comment by XuNeo** [comment]
> Is Kconfig able to cover our use case? I'm not aware of a mode where we are able to generate a "full list" of the macro definitions with some default values being overwritten.

What Kconfig can't do is configurations needs preprocessor. Other than that, it works great and has lots of tools. RTOS NuttX,  Zephyr, RT-Thread etc and linux all support Kconfig.

For preprocessor, the solution in NuttX is to define it in Makefile.
See 
https://github.com/apache/nuttx-apps/blob/7c4590b81801bde3cfd7f2fd0919c01c41a78024/graphics/lvgl/Makefile#L30-L32

**Comment by kisvegabor** [comment]
I'd like to go with a simple and platform independent solution first. I think it's the safest bet. We can improve it later if needed.

**Review by kisvegabor — APPROVED** [review]

**Review by liamHowatt — APPROVED** [review]

**Pull request merged**